internal/repository/targets.go

```diff
@@ -33,15 +29,12 @@ func (r *Repository) InitializeTargets(ctx context.Context, targetsKeyBytes []by
 		return ErrCannotReinitialize
 	}
 
-	// Verify if targetsKeyBytes is authorized to sign for the role
-	authorizedKeyIDs, err := state.FindAuthorizedSigningKeyIDs(ctx, targetsRoleName)
-	if err != nil {
-		return err
-	}
-
-	if !isKeyAuthorized(authorizedKeyIDs, keyID) {
-		return ErrUnauthorizedKey
-	}
+	// TODO: verify is role can be signed using the presented key. This requires
+	// the user to pass in the delegating role as well as we do not want to
+	// assume which role is the delegating role (diamond delegations are legal).
+	// See: https://github.com/theupdateframework/specification/issues/19,
+	// https://github.com/theupdateframework/specification/issues/214, and
+	// https://github.com/theupdateframework/python-tuf/issues/660.
```

Comment:
IIUC, this is the main bit of code that isn't a simple refactor / rename, and it looks like we're removing some verification logic, so it's non-obvious to me how this is avoiding the promiscuous delegation verification issue (though I'm not 100% sure I completely understand the scenario we're looking to avoid 😓).

Reply:
Yes! Let me clarify. Until this PR, if you tried to update gittuf policy metadata `some-rules.json` with key `A`, gittuf would determine if key `A` is trusted to sign `some-rules.json` using the delegations graph. However, diamond delegations are legal in TUF, where role `1` and role `2` can both delegate to `some-rules`. However, role `1` may say `some-rules` must be signed by key `B` while role `2` may say `some-rules` must be signed by key `A`. When signing `some-rules` we don't know which delegation applies implicitly.

So, this change removes that check, and the TODO is to present the user with some way of explicitly specifying the delegation they're signing `some-rules` for, as an added sanity check. This does not downgrade the security itself because the validity of `some-rules`'s signatures is now checked during the actual verification process for some namespace. In that case, gittuf knows how it reached `some-rules` because we have a specific namespace to walk the graph for. So, gittuf knows it reached `some-rules` via role `2`, and therefore key `A` is trusted. OTOH, `some-rules` would not be trusted (in this case) if it is reached via role `1` which expects `some-rules` to be signed by key `B`. There's possibly a different (TUF-scoped) discussion about whether these kinds of delegations should be legal, but this change aligns gittuf to how role signatures must be validated in TUF right now.

Reply:
https://github.com/gittuf/gittuf/issues/246 captures this, so I'll update the inline TODO as well.